### PR TITLE
VMware: datastore selection

### PR DESCRIPTION
**Pull request by mbusbee**
##### SUMMARY
This fix adds additional parameter 'datastore'. This parameter
provides a flexibility to specify custom datastore or datastore cluster
to deploy new virtual machine from template (which is located in different
datastore or datastore cluster from virtual machine's datastore or datastore
cluster).

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME

lib/ansible/modules/cloud/vmware/vmware_guest.py

##### ANSIBLE VERSION

```
2.6devel
```

### CODE REVIEW

**Comment by Akasurde** [comment]
@pdellaert @tchernomax Could you please review this ?

File: lib/ansible/modules/cloud/vmware/vmware_guest.py [hunk 1]
```diff
@@ -263,6 +263,14 @@
     - ' - C(type) (string): Value type, string type by default.'
     - ' - C(operation): C(remove): This attribute is required only when removing properties'
     version_added: '2.6'
+  datastore:
+    description:
+    - Specify datastore or datastore cluster to provision virtual machine.
+    - 'This will take precendence over "disk.datastore" parameter.'
```
Comment (pdellaert) [inline lib/ansible/modules/cloud/vmware/vmware_guest.py:331]: Does this mean that setting the datastore for each disk separately will no longer work?

Comment (Akasurde) [reply lib/ansible/modules/cloud/vmware/vmware_guest.py:331]: Currently, specifying separate datastore in disk is not working either.

https://github.com/ansible/ansible/blob/87d6bdaf9819fdc45a5d309e5e54f8b6f6651ada/lib/ansible/modules/cloud/vmware/vmware_guest.py#L1442

**Comment by Akasurde** [comment]
@dericcrago Could you please review this ?

**Comment by dericcrago** [comment]
I don't see any tests in this PR, are there already tests for this functionality?

**Comment by Akasurde** [comment]
@dericcrago This requires creating a template in different datastore or datastore cluster. VCSIM does not support template at this moment. So no testcase for this scenario. Let me know if you have any idea or suggestion to test this feature.

**Comment by corbinstuard** [comment]
Is this change still WIP? It is very much desired.

**Comment by Akasurde** [comment]
@corbinstuard I am waiting for the reviews. I would really appreciate if you or someone else could test this and provide feedback. Thanks.

**Comment by moshloop** [comment]
I have tested and confirmed that this works on vSphere 6.5 cloning into a different cluster & datastore.

**Comment by joerussbowman** [comment]
I think I'm waiting on this as well, unless there is another way to deploy a template to another host/datastore? I've tried specifying esxi_hostname and a datastore on the template disk definition. Is this still being reviewed?

**Comment by moshloop** [comment]
@Akasurde what else is required to get this merged?

**Comment by tomtastic** [comment]
Can this be backported to 2.4, I hit https://github.com/ansible/ansible/issues/35012 this morning with Ansible 2.4.2.0

**Comment by Akasurde** [comment]
@tomtastic I am afraid that 2.4 is only open for [security fixes](https://docs.ansible.com/ansible/latest/reference_appendices/release_and_maintenance.html#release-status) and we can not backport this under that criteria. I would recommend you to update Ansible to 2.7. Thanks.

**Pull request merged**